lib/datadog/appsec/configuration/settings.rb

```diff
@@ -95,6 +95,62 @@ def self.add_settings!(base)
                 end
               end
 
+              settings :block do
+                # HTTP status code to block with
+                option :status do |o|
+                  o.default { 403 }
+                end
+
+                # only applies to redirect status codes
+                option :location do |o|
+                  o.setter { |v| URI(v) unless v.nil? }
+                end
+
+                # only applies to non-redirect status codes with bodies
+                option :templates do |o|
+                  o.default do
+                    json = ENV.fetch(
+                      'DD_APPSEC_HTTP_BLOCKED_TEMPLATE_JSON',
+                      :json
+                    )
+
+                    html = ENV.fetch(
+                      'DD_APPSEC_HTTP_BLOCKED_TEMPLATE_HTML',
+                      :html
+                    )
+
+                    text = ENV.fetch(
+                      'DD_APPSEC_HTTP_BLOCKED_TEMPLATE_TEXT',
+                      :text
+                    )
+
+                    {
+                      'application/json' => json,
+                      'text/html' => html,
+                      'text/plain' => text,
+                    }
+                  end
+                  o.setter do |v|
+                    next if v.nil?
+
+                    # TODO: should merge with o.default to allow overriding only one mime type
```

Comment:
Without this, if `templates` is set it overrides the whole content, e.g the following removes `text/plain` and `application/json`

```
    c.appsec.block.templates = {
      'text/html' => 'block.html',
    }
```